library/std/src/panicking.rs

```diff
@@ -725,20 +723,17 @@ fn rust_panic_with_hook(
             panic_count::MustAbort::AlwaysAbort => {
                 // Unfortunately, this does not print a backtrace, because creating
                 // a `Backtrace` will allocate, which we must to avoid here.
-                let panicinfo = PanicInfo::internal_constructor(
-                    message,
-                    location,
-                    can_unwind,
-                    force_no_backtrace,
-                );
-                rtprintpanic!("{panicinfo}\npanicked after panic::always_abort(), aborting.\n");
+                if let Some(message) = message {
+                    rtprintpanic!("aborting due to panic at {location}:\n{message}\n");
+                } else {
+                    rtprintpanic!("aborting due to panic at {location}\n");
+                }
```

Comment:
Should this mention `always_abort`? Could be hard to figure out otherwise why is aborts.

Reply:
`always_abort` isn't stable. The only stable way users can run into this error is through `Command::spawn` with `Command::pre_exec`. I don't think adding `always_abort` in the note would make it easier to trace it down to `Command`.

Reply:
It would at least provide a clue to search for in the standard library source code.

Reply:
> It would at least provide a clue to search for in the standard library source code.

If you're going to search through the stdlib code, then just searching for the message itself also works perfectly fine. ^^

I can't think of a real life scenario (in stable Rust) where mentioning the internal/unstable name `always_abort` makes things clearer rather than just a distraction. If there is a panic in a `pre_exec` hook, the problem is not that `always_abort` was called internally; instead, the problem is that there was a panic, so I believe it's better to focus on that.

Reply:
> If you're going to search through the stdlib code, then just searching for the message itself also works perfectly fine. ^^

The current panic message is so generic I wouldn't be surprised if it would be used at multiple locations in the future. It also isn't useful if you already know about always_abort but don't recall the exact panic message for it. Even just mentioning that we are at a place where panics are not allowed would be an improvement.

Reply:
I really don't think we should be showing unstable internal implementation details (such as always_abort) in messages that are shown to end users of Rust programs.